SCons/Tool/ninja/__init__.py

```diff
@@ -65,8 +65,11 @@ def ninja_builder(env, target, source):
     print("Generating:", str(target[0]))
 
     generated_build_ninja = target[0].get_abspath()
-    NINJA_STATE.generate()
-
+    try:
+        NINJA_STATE.generate()
+    except Exception as e:
+        print(f"ERROR: an excetion occurred while generating the ninja file:\n{traceback.format_exc()}")
```

Comment:
Should probably cause this to propagate to SCons's exception handling?

Reply:
yep, updated